internal/storage/sbom_store.go

```diff
@@ -0,0 +1,525 @@
+	if err != nil {
+		return storage.NewInternalError(err.Error())
+	}
+	defer func() {
+		if err := tx.Rollback(); !errors.Is(err, sql.ErrTxDone) {
+			log.Printf("failed to rollback transaction: %v", err)
+		}
+	}()
+
+	query, args, err := sq.Delete("sboms").
+		Where(sq.Eq{"name": name, "namespace": namespace}).
+		Suffix("RETURNING *").
+		ToSql()
+	if err != nil {
+		return storage.NewInternalError(err.Error())
+	}
+
+	sbomRecord := &sbomSchema{}
+	if err := tx.GetContext(ctx, sbomRecord, query, args...); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return storage.NewKeyNotFoundError(key, 0)
+		}
+		return storage.NewInternalError(err.Error())
+	}
+
+	if err := json.Unmarshal(sbomRecord.Object, out); err != nil {
+		return storage.NewInternalError(err.Error())
+	}
+
+	if err := preconditions.Check(key, out); err != nil {
+		return err
+	}
+
+	if err := validateDeletion(ctx, out); err != nil {
+		return err
+	}
+
+	if err := tx.Commit(); err != nil {
+		return storage.NewInternalError(err.Error())
+	}
+
+	if err := s.broadcaster.Action(watch.Deleted, out); err != nil {
+		return storage.NewInternalError(err.Error())
+	}
+
+	return nil
+}
+
+// Watch begins watching the specified key. Events are decoded into API objects,
+// and any items selected by 'p' are sent down to returned watch.Interface.
+// resourceVersion may be used to specify what version to begin watching,
+// which should be the current resourceVersion, and no longer rv+1
+// (e.g. reconnecting without missing any updates).
+// If resource version is "0", this interface will get current object at given key
+// and send it in an "ADDED" event, before watch starts.
+func (s *sbomStore) Watch(ctx context.Context, key string, opts storage.ListOptions) (watch.Interface, error) {
+	if opts.ResourceVersion == "" {
+		return s.broadcaster.Watch()
+	}
+
+	if opts.ResourceVersion == "0" {
+		sbom := &v1alpha1.SBOM{}
+		if err := s.Get(ctx, key, storage.GetOptions{}, sbom); err != nil {
+			return nil, err
+		}
+
+		return s.broadcaster.WatchWithPrefix([]watch.Event{{Type: watch.Added, Object: sbom}})
+	}
+
+	sbomList := &v1alpha1.SBOMList{}
+	if err := s.GetList(ctx, key, storage.ListOptions{}, sbomList); err != nil {
+		return nil, err
+	}
+	var events []watch.Event
+	for _, item := range sbomList.Items {
+		events = append(events, watch.Event{Type: watch.Added, Object: &item})
+	}
+
+	return s.broadcaster.WatchWithPrefix(events)
+}
+
+// Get unmarshals object found at key into objPtr. On a not found error, will either
+// return a zero object of the requested type, or an error, depending on 'opts.ignoreNotFound'.
+// Treats empty responses and nil response nodes exactly like a not found error.
+// The returned contents may be delayed, but it is guaranteed that they will
+// match 'opts.ResourceVersion' according 'opts.ResourceVersionMatch'.
+func (s *sbomStore) Get(ctx context.Context, key string, opts storage.GetOptions, objPtr runtime.Object) error {
```

Comment:
It looks like we are not taking into account the selectors that might be provided by the user

Reply:
The `Get` call doesn't use selectors.